src/fheroes2/maps/maps_tiles.cpp

```diff
@@ -1752,19 +1752,24 @@ void Maps::Tiles::CaptureFlags32( const MP2::MapObjectType objectType, int col )
     }
 }
 
-void Maps::Tiles::CorrectFlags32( u32 index, bool up )
+void Maps::Tiles::CorrectFlags32( int col, u32 index, bool up )
 {
-    TilesAddon * taddon = FindFlags();
-
-    // replace flag
-    if ( taddon )
-        taddon->index = index;
-    else if ( up )
-        // or new flag
-        addons_level2.emplace_back( TilesAddon::UPPER, World::GetUniq(), 0x38, index );
-    else
-        // or new flag
-        addons_level1.emplace_back( TilesAddon::UPPER, World::GetUniq(), 0x38, index );
+    if ( col == Color::NONE ) {
+        removeFlags();
+    }
```

Comment:
```suggestion
    if ( col == Color::NONE ) {
        removeFlags();
        return;
    }
```
And revert `else` branch whitespace changes. Patch will be smaller.

Reply:
Maybe...